cmake/hwloc_detection.cmake

```
@@ -51,9 +51,9 @@ if (NOT HWLOC_TARGET_EXPLICITLY_DEFINED AND
     NOT TBB_DISABLE_HWLOC_AUTOMATIC_SEARCH
 )
     find_package(PkgConfig QUIET)
-    if (PKG_CONFIG_FOUND)
-        pkg_search_module(HWLOC hwloc)
-        if (HWLOC_FOUND)
+    if (PKG_CONFIG_FOUND AND CMAKE_VERSION VERSION_GREATER 3.6)
```

`VERSION_GREATER`  will lead to `>3.6`, but you need `>=3.6`. CMake provides `VERSION_GREATER_EQUAL` but it is not available in CMake 3.1. So, please consider this:
```suggestion
    if (PKG_CONFIG_FOUND AND NOT CMAKE_VERSION VERSION_LESS 3.6)
```